src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordDeconstruct.cs

```diff
@@ -59,6 +61,27 @@ protected override (TypeWithAnnotations ReturnType, ImmutableArray<ParameterSymb
 
         protected override int GetParameterCountFromSyntax() => _ctor.ParameterCount;
 
+        protected override void MethodChecks(BindingDiagnosticBag diagnostics)
+        {
+            base.MethodChecks(diagnostics);
+
+            if (ParameterCount == _properties.Length)
+            {
+                for (int i = 0; i < _properties.Length; i++)
+                {
+                    var property = _properties[i];
+                    NamedTypeSymbol containingType = this.ContainingType;
+
+                    if (property.ContainingType != (object)containingType &&
+                        !containingType.GetMembers(property.Name).IsEmpty)
+                    {
+                        // The positional member was inherited but is hidden by a member of the current record type
+                        diagnostics.Add(ErrorCode.ERR_HiddenPositionalMember, _parameters.Parameters[i].Identifier.GetLocation(), property);
```

Comment:
I think this isn't the right place to detect and report the error. I think this should be done when we are looking for positional members and regardless whether the "Deconstruct" is explicitly provided.